packages/tsup-configs/src/dev.ts

```diff
@@ -0,0 +1,18 @@
+import { type DefineConfigOptions, defineConfigBase } from "./defineConfigBase.ts";
+
+
+export function defineDevConfig(config?: DefineConfigOptions) {
+    return defineConfigBase(config, {
+        watch: true,
+        dts: {
+            compilerOptions: {
+                noUnusedLocals: false
```

Comment:
Sorry to get back to this again but I really don't get our current stance :)

So the root of the issue here is that we choosed to set in our `tsconfig` the field `noUnusedLocals` to `true`, which is false by default.

I am not sure why we did this as we already have ESLint with the [no-unused-vars](https://eslint.org/docs/latest/rules/no-unused-vars) which help us with this.

As we lint locally and on our pipelines it just seems overcomplicated and redundent to also activate those rules (which are not activated by default) in our `tsconfig` as well.

It's clear why they have been added to `tsconfig`, my feeling is that it's for usecase where you don't want to use ESLint or any other linter in combination with `tsc`.

Rules like that won't change anything to the compiled code, it's mostly to improve the code quality. In those case, we already have ESLint covering us.

Reply:
So, IMO, we should remove `noUnusedLocals` and any other linting rules already covered by ESLint from our `tsconfig` and then remove the `compilerOptions` in this tsup config.